lib/Infer/EnumerativeSynthesis.cpp

```diff
@@ -207,7 +217,7 @@ void getGuesses(std::vector<Inst *> &Guesses,
   Comps.push_back(I2);
 
   for (auto K : BinaryOperators) {
-    if (Inst::isCmp(K) && Width != 1)
+    if ((Inst::isCmp(K) || Inst::isOverflowIntrinsicSub(K)) && Width != 1)
```

Comment:
wait, why only subtract here?

Reply:
This is not subtract. This is "Sub" component which refers to "AddO" variants. See this comment: https://github.com/google/souper/pull/598/files/ac7ad6e88b7fba740b87a1a41c265f6e9537a8e3#diff-a263ca9d697aae31475ef84f4ec04418R51-R54 for explanation how these are treated differently.